### PR TITLE
CVSL-1291 add offender number as a case identifier to emails

### PR DESCRIPTION
**Pull request by louisenorris**
Added `offenderNo` as a case identifier to pass to notify emails in HDC. Templates will need to be updated in Notify but have tested changes work in a new test template in the service. Also put together a basic flow diagram of how the notification functionality is structured in the service to help me do this and included this below in case useful for the PR:

HDC email notifications

From what I can see the `sendReminder` and `sendReminders` functions in the `reminderService` aren't actually used anywhere(?) so might remove them in the future but didn't want to risk doing this now as part of this work.

### CODE REVIEW

File: server/routes/send.js [hunk 1]
```diff
@@ -29,11 +29,13 @@ module.exports =
       asyncMiddleware(async (req, res) => {
         const { destination, bookingId } = req.params
         const { token, licence, prisoner } = res.locals
+        const { offenderNo } = prisoner
         const transition = transitionsForDestinations[destination]
 
         await notificationService.send({
           transition,
           bookingId,
+          offenderNo,
```
Comment (andrewrlee) [inline server/routes/send.js:38]: is the offenderNo not already passed through as part of the prisoner below?  
Maybe you don't need to pass it separately?

File: server/services/notifications/roNotificationHandler.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ module.exports = function createRoNotificationHandler(
   }
 
   return {
-    async sendRo({ transition, bookingId, token, user }) {
+    async sendRo({ transition, bookingId, offenderNo, token, user }) {
```
Comment (andrewrlee) [inline server/services/notifications/roNotificationHandler.js:88]: prisoner is passed through to here, its just not being deconstructed

File: server/routes/send.js (hunk 1 above)
Comment (louisenorris) [reply server/routes/send.js:38]: Yep - reverted this in latest commit

File: types/licences.d.ts [hunk 3]
```diff
@@ -43,9 +43,20 @@ export interface RoContactDetailsService {
...
 
+interface PrisonerDetails {
+  firstName: string
+  lastname: string
+  offenderNo: stiring
+}
+
+interface Prisoner extends PrisonerDetails {
+  any
+}
```
Comment (louisenorris) [inline types/licences.d.ts:54]: Not sure if this is typescript specific enough but the whole prisoner object looks quite extensive and contains another nested objects so wondering if this is ok to just make sure we have type safety on the prisoner parameters we require for the emails?

Comment (andrewrlee) [reply types/licences.d.ts:54]: this doesn't look like valid typescript to me?

Yes, its fine just to add the fields you are interested in.. not sure why you need the separate interface though?  Could you just have one?

Comment (louisenorris) [reply types/licences.d.ts:50]: Thanks, corrected this in latest commit

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**